### DISCUSSION TITLE
Keenetic cloud authorised web access not working

### DISCUSSION THREAD

**Discussion by Seziksezik** [emsesp/EMS-ESP32#1896] [Q&A]
Problem with authorized access trough keenetic cloud.

Keenetic is.... a  former part Zyxell 
We have there big keenetic routers installed base. 

This company is very close to Zyxell..

So... they have a cloud....For example. I.can access to my home network with easy address. 

http://ems.blablabla.keenetic.pro

this is service description:

https://help.keenetic.com/hc/en-us/articles/360008818279-Keenetic-Cloud-web-portal

Difficult repeat..Need access to Keenetic cloud. But I can add read only account to check it. 

Login/password are default..

May be will be the same problem with ddns service or other who support user authorisation for remote access. 

It works directly from my browser without any authorisation. But.... If I try set up login/password based access... everything not working. 

I can see only login page :) Enter  login/password for.the cloud and see an error message..

I think there are no delay somewhere. See attached screen shots. 

![Screenshot_20240724_213121_Browser](https://github.com/user-attachments/assets/5fa70f68-8235-452d-9f47-06490bf9cbb1)
![Screenshot_20240724_214434_Browser](https://github.com/user-attachments/assets/9f3a283a-e81e-4d23-b2a2-f796d4ca6c5d)
![Screenshot_20240724_220623_Browser](https://github.com/user-attachments/assets/60da7d5a-c4aa-45bc-b3cc-6c70eae4a3d5)

**Comment by proddy** [comment]
turn on the browsers dev console (F12) and see what errors are reporting after you click login. It will help us pin point the problem. You may also need to enable CORS in ems-esp.

**Comment by Seziksezik** [comment]
A question. How to enable cors?:) 

Look my video on Google Chrome. 
I don't have enough time to enter password. Windows closed in 10 seconds around. Yes, I have access but see almost nothing. 

And F12 screen photo. 

I used first time  a yandex browser. From my smartphone. Don't know how to press F12 there. 

3d small screen copy is Chrone from my Smartphone. I enter the password then see network timeout. 

Without password protection it works. 


https://github.com/user-attachments/assets/59b9a8ee-2884-4861-b84b-c41b578e7aa1



@
![Screenshot_20240725_152708_Gallery](https://github.com/user-attachments/assets/d04f0786-fb9d-4639-8f16-c5850f87ec13)

**Comment by Seziksezik** [comment]
![Screenshot_20240725_153328_Chrome](https://github.com/user-attachments/assets/611475b6-ba5f-406f-887e-709200c2eab9)

3d file. Smartphone. Google Chrome. Password entered correctly.

**Comment by proddy** [comment]
- what do you mean by "without password protection, it works"? Is this something on your VPN
- CORS can be enabled in EMS-ESPs Network Settings
- When you're in F12, go the Network tab and then CTRL-R. Then we can see all the traffic. Something is being blocked.

**Comment by Seziksezik** [comment]
look. I have two option when access to remote site. With password and without password. If without password - it works. 

But if I switch on password protection something not working. See details.. @

https://github.com/user-attachments/assets/08c21b1a-6625-4168-bf06-8b1b548d50b7

![20240725_155731](https://github.com/user-attachments/assets/57d8d85b-7e76-4d80-bd0d-e07c836f2087)
![Screenshot_20240725_160054_Gallery](https://github.com/user-attachments/assets/449107cb-3c20-46b5-a4da-d59d4de40f6c)
![Screenshot_20240725_160049_Gallery](https://github.com/user-attachments/assets/2a414247-d429-4286-87d3-5d5e4ecdce55)

**Comment by proddy** [comment]
click on one of those red signIn errors (where it says cancelled). then in the tab Headers and also Response show me what you see. Like:

<img width="657" alt="image" src="https://github.com/user-attachments/assets/94b8a79c-ea28-44a9-8153-d7c1f45643f4">

**Comment by proddy** [comment]
also do you have examples of other ESP based applications where it does work, like Tasmota, ESPHome ? I'm thinking its either because we use HTTP and not HTTP/S or there's an HTTP header missing in EMS-ESP that you need to authorize the client

**Comment by Seziksezik** [comment]
see attached. 

Yes. Cloud use https protocol. 

![Screenshot_20240725_163800_Gallery](https://github.com/user-attachments/assets/7d1857e0-1db4-4780-b62e-1737118c1e35)

![Screenshot_20240725_163902_Gallery](https://github.com/user-attachments/assets/fc73ecd3-ecf2-4222-977d-6fdceb4b220f)

Sorry. I dont have  Tasmota or ESP Home  Just Tuya :) 

I can access remote to the device, but need start SSTP client, then put IP address. Not a problem. But looks like it is not compatible with keenetic cloud. urls based access is more easy.

**Comment by Seziksezik** [comment]
So strange. 

I can enter inside troughbthe cloud when use the link lice... https://ems.sezik.keenetic.pro/devices

Then enter cloud login/password, device login password... and see something. 

![Screenshot_20240725_173937_Gallery](https://github.com/user-attachments/assets/d0ab64d5-3bf6-4c80-b3dc-4153ed1a8e4d)
![Screenshot_20240725_173944_Gallery](https://github.com/user-attachments/assets/825b1c54-3d1b-48ac-bd77-3d17bafeed79)

See attached picture.

**Comment by proddy** [comment]
I can see none of the backend API calls are working (they all show as red and cancelled). I don't know what is causing it, sorry. It may be the bearer token used in the authentication. Can you try if this works `http://<IP of EMS-ESP>/api/system`. It should bring back some JSON data. If that works, next test is using Curl with some headers as in the examples on https://emsesp.github.io/docs/Commands/#via-the-command-line

**Comment by Seziksezik** [comment]
ok. no problem. 
![20240725_182123](https://github.com/user-attachments/assets/e77b8732-7a91-410d-a9a9-0d7ea1b6ff2b)
Anyway thank you for your support. I think it is cloud problem somewhere. 

See direct link access.

**Comment by proddy** [comment]
Okay. I have EMS-ESP running in a VPN/DDNS in Cloudflare and it works, but nothing as sophisticated as Keenetic. I'll turn this into a discussion.

**Comment by AlexeySpiridonov** [comment]
You trying to use KeenDNS for remote access to third party solutions. But KeenDNS created only for remote access to Keenetic admin interface.   Other use cases usually work, but Keenetic does not guarantee the operation of web interfaces on other devices via KeenDNS.

можете считать это официальным ответом от кинетика
